packages/server-wallet/src/models/channel.ts

```diff
@@ -173,6 +177,67 @@ export class Channel extends Model implements RequiredColumns {
     });
   }
 
+  public static directFundingStatus(
+    supported: SignedStateVarsWithHash | undefined,
+    fundingFn: (address: Address) => ChannelStateFunding,
+    myParticipant: Participant,
+    fundingStrategy: FundingStrategy
+  ): FundingStatus {
+    if (fundingStrategy !== 'Direct') {
+      return 'Uncategorized';
+    }
+
+    const outcome = supported?.outcome;
+    if (!supported || !outcome) {
+      return 'Uncategorized';
+    }
+
+    const {allocationItems, assetHolderAddress} = checkThat(outcome, isSimpleAllocation);
+
+    // Collapse all allocation items with my destination into one
+    const myAllocation = allocationItems
```

Comment:
This would probably be easier to read if you defined
```
const myDestination = myParticipant.destination;
const myAllocation = allocationItems.filter(**snip**).map(ai => ai.amount).reduce(BN.add, BN.from(0));
```

Reply:
Good call 7fd576927